frame/contracts/src/lib.rs

```diff
@@ -1287,4 +1162,12 @@ where
 	fn min_balance() -> BalanceOf<T> {
 		<T::Currency as Inspect<AccountIdOf<T>>>::minimum_balance()
 	}
+
+	/// Convert a 1D Weight to a 2D weight.
+	///
+	/// Used by backwards compatible extrinsics. We cannot just set the proof to zero
+	/// or an old `Call` will just fail.
+	fn compat_weight(gas_limit: OldWeight) -> Weight {
+		Weight::from(gas_limit).set_proof_size(u64::from(T::MaxCodeLen::get()) * 2)
```

Comment:
Dumb question: Is the `MacCodeLen * 2` always a sufficient approximation of gas?

Reply:
Depends on how many contract you call and how big they are in a single transaction. This is just a ad-hoc solution to support in-storage `Call`s that might exist in a `Scheduler` of some runtime. This is very niche. You should use the dry-run to come up with a proper value and don't rely on this.